### PR TITLE
12.0 mig sale order line input

### PR DESCRIPTION
**Pull request by mpanarin**
migration to v 12.0

### CODE REVIEW

File: sale_order_line_input/views/sale_order_line_view.xml [hunk 1]
```diff
@@ -0,0 +1,102 @@
+<?xml version="1.0"?>
+<!-- Copyright 2018 Tecnativa - Carlos Dauden
+     License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). -->
+<odoo>
+  <record id="view_sales_order_line_input_tree" model="ir.ui.view">
+    <field name="name">sale.order.line.input.tree.</field>
+    <field name="model">sale.order.line</field>
+    <field name="priority">20</field>
+    <field name="arch" type="xml">
+      <tree string="Sales Orders Lines"
```
Comment (rven) [inline sale_order_line_input/views/sale_order_line_view.xml]: Deprecated <tree> xml attribute "string" (xml-deprecated-tree-attribute)
```suggestion
      <tree
```

File: sale_order_line_input/readme/USAGE.rst [hunk 2]
```diff
@@ -0,0 +1,6 @@
+To use this module, you need to:
+
+#. Go to Sales > Orders > Sales Orders Lines
+#. You can filter, group and switch view type
+#. Create new line related with existant order
```
Comment (rven) [inline sale_order_line_input/readme/USAGE.rst]:
```suggestion
#. Create new line related with existing order
```

File: sale_order_line_input/views/sale_order_line_view.xml [hunk 3]
```diff
@@ -0,0 +1,102 @@
+<?xml version="1.0"?>
+<!-- Copyright 2018 Tecnativa - Carlos Dauden
+     License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). -->
+<odoo>
+  <record id="view_sales_order_line_input_tree" model="ir.ui.view">
+    <field name="name">sale.order.line.input.tree.</field>
```
Comment (rven) [inline sale_order_line_input/views/sale_order_line_view.xml:6]:
```suggestion
    <field name="name">sale.order.line.input.tree</field>
```

File: sale_order_line_input/views/sale_order_line_view.xml [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+      <tree string="Sales Orders Lines"
+            decoration-info="invoice_status=='to invoice'" decoration-warning="state=='draft'" decoration-muted="state=='cancel'" decoration-success="state=='done'"
+            editable="top" create="true">
+        <field name="order_id" string="Order"
+               required="0" placeholder="New"
+               domain="[('state', 'not in', ('done', 'cancel'))]"
+               attrs="{'readonly': [('order_id', '!=', False)]}"
+               force_save="1"
+               options='{"no_open": True, "no_create": True}'/>
+        <button name="action_sale_order_form" type="object" icon="fa-external-link" class="oe_stat_button"/>
```
Comment (rven) [inline sale_order_line_input/views/sale_order_line_view.xml]:
```suggestion
        <button name="action_sale_order_form" string="Open Sale Order" type="object" icon="fa-external-link" class="oe_stat_button"/>
```

File: sale_order_line_input/views/sale_order_line_view.xml [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+               string="Ordered Qty"
+               context="{'partner_id':order_partner_id, 'quantity':product_uom_qty, 'pricelist':pricelist_id, 'uom':product_uom, 'company_id': company_id}"
+               attrs="{'readonly': [('state', 'in', ('done', 'cancel'))]}"/>
+        <field name="qty_delivered" sum="Delivered" attrs="{'readonly': [('qty_delivered_method', '!=', 'manual')]}"/>
+        <field name="qty_invoiced" sum="Invoiced"/>
+        <field name="qty_to_invoice" sum="To Invoice"/>
+        <field name="product_uom"
+               attrs="{'readonly': [('state', 'in', ('sale','done', 'cancel'))]}"
+               context="{'company_id': company_id}"
+               groups="product.group_uom" options='{"no_open": True}'/>
```
Comment (rven) [inline sale_order_line_input/views/sale_order_line_view.xml]:
```suggestion
               groups="uom.group_uom" options='{"no_open": True}'/>
```

File: sale_order_line_input/views/sale_order_line_view.xml [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+    </field>
+  </record>
+
+  <record id="view_sales_order_line_filter" model="ir.ui.view">
+    <field name="model">sale.order.line</field>
+    <field name="inherit_id" ref="sale.view_sales_order_line_filter"/>
+    <field name="arch" type="xml">
+      <xpath expr="//field[@name='salesman_id']" position="after">
+        <field name="name"/>
+        <field name="product_uom" groups="product.group_uom"/>
```
Comment (rven) [inline sale_order_line_input/views/sale_order_line_view.xml]:
```suggestion
        <field name="product_uom" groups="uom.group_uom"/>
```

File: sale_order_line_input/views/sale_order_line_view.xml [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+        <field name="analytic_tag_ids" groups="analytic.group_analytic_accounting"/>
+        <separator/>
+        <filter name="not_invoice_related" string="Not Invoice Related" domain="[('invoice_lines', '=', False)]"/>
+        <filter name="invoice_related" string="Invoice Related" domain="[('invoice_lines', '!=', False)]" context="{'tax_hide': True}"/>
+      </xpath>
+      <xpath expr="//filter[@name='product']/.." position="inside">
+        <filter name="order_state" string="Order State" context="{'group_by': 'state'}"/>
+        <filter name="invoice_status" string="Invoice Status" context="{'group_by': 'invoice_status'}"/>
+        <filter name="uom_filter" string="UoM" context="{'group_by': 'product_uom'}" groups="product.group_uom"/>
```
Comment (rven) [inline sale_order_line_input/views/sale_order_line_view.xml]:
```suggestion
        <filter name="uom_filter" string="UoM" context="{'group_by': 'product_uom'}" groups="uom.group_uom"/>
```

File: sale_order_line_input/views/sale_order_line_view.xml [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+               groups="analytic.group_analytic_accounting" widget="many2many_tags" options="{'color_field': 'color'}"/>
+        <field name="price_unit"
+               attrs="{'readonly': ['|', ('qty_invoiced', '&gt;', 0), ('state', 'in', ('done', 'cancel'))]}"/>
+        <field name="tax_id" widget="many2many_tags" options="{'no_create': True}" domain="[('type_tax_use','=','sale'),('company_id','=',company_id)]"
+               invisible="context.get('tax_hide', False)"
+               attrs="{'readonly': ['|', ('qty_invoiced', '&gt;', 0), ('state', 'in', ('done', 'cancel'))]}"/>
+        <field name="discount"
+               attrs="{'readonly': [('state', 'in', ('done', 'cancel'))]}"
+               groups="sale.group_discount_per_so_line"/>
+        <field name="price_subtotal" sum="Total" widget="monetary" groups="sale.group_show_price_subtotal"/>
```
Comment (rven) [inline sale_order_line_input/views/sale_order_line_view.xml]:
```suggestion
        <field name="price_subtotal" sum="Total" widget="monetary" groups="account.group_show_line_subtotals_tax_excluded"/>
```

File: sale_order_line_input/views/sale_order_line_view.xml [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+        <field name="price_total" widget="monetary" groups="sale.group_show_price_total"/>
```
Comment (rven) [inline sale_order_line_input/views/sale_order_line_view.xml]:
```suggestion
        <field name="price_total" widget="monetary" groups="account.group_show_line_subtotals_tax_included"/>
```

**Review by rven — CHANGES_REQUESTED** [review]

**Comment by rven** [comment]
The changes to get it to work can be found here
https://github.com/mpanarin/sale-workflow/pull/1

**Review by oihane — APPROVED** [review]
LGTM :+1:

**Comment by anajuaristi** [comment]
LGTM :+1:

**Review by Tardo — APPROVED** [review]
LGTM :+1: code review

**Pull request closed without merging**